### PR TITLE
remove unneeded AB3 code

### LINKED ISSUE #20: Cleanup outdated AB3 code

### PR DESCRIPTION
**Pull request by aqcd**
resolves #20 

also adds some tests

### CODE REVIEW

**Comment by aqcd** [comment]
Pushed to future PRs:
1. LocationList current defaults to Optional.empty, to be fixed and tested.
2. RecipeParserUtilTest incomplete, does not test for ingredient-related parsing.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -1,124 +0,0 @@
...
-import seedu.address.commons.util.StringUtil;
-import seedu.address.logic.parser.exceptions.ParseException;
-import seedu.address.model.person.Address;
-import seedu.address.model.person.Email;
-import seedu.address.model.person.Name;
-import seedu.address.model.person.Phone;
-import seedu.address.model.tag.Tag;
-
-/**
- * Contains utility methods used for parsing strings in the various *Parser classes.
```
Comment (blackonyyx) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:19]: We are replacing this with parser level specific methods? 
how about item name etc
Or issit gonna use RegEx instead?

File: src/main/java/seedu/address/ui/UiManager.java [hunk 2]
```diff
@@ -23,7 +23,7 @@
     private static final String ICON_APPLICATION = "/images/address_book_32.png";
```
Comment (blackonyyx) [inline src/main/java/seedu/address/ui/UiManager.java]: could probably rename -> inventoryinator.png 
or something like that, to remove any reference to ab3

File: src/test/java/seedu/address/logic/commands/ModelStub.java [hunk 3]
```diff
@@ -7,19 +7,18 @@
...
 import seedu.address.model.ReadOnlyUserPrefs;
 import seedu.address.model.item.Item;
 import seedu.address.model.location.Location;
-import seedu.address.model.person.Person;
 import seedu.address.model.recipe.Recipe;
 
 /**
- * A default model stub that have all of the methods failing EXCEPT processPrecursors.
- * This is because we minimally need the processPrecursors method to handle AddItemCommand.
+ * A default model stub that have all of the methods failing EXCEPT processPrecursors and getFilteredItemList.
```
Comment (blackonyyx) [inline src/test/java/seedu/address/logic/commands/ModelStub.java:19]: Im not so certain that getFilteredItemList is the only list we need for a ModelStub
How about the recipe list or location list?

File: src/test/java/seedu/address/storage/JsonSerializableRecipeListTest.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    private static final Path INVALID_RECIPE_FILE = TEST_DATA_FOLDER.resolve("invalidRecipeRecipeList.json");
+    private static final Path DUPLICATE_RECIPE_FILE = TEST_DATA_FOLDER.resolve("duplicateRecipeRecipeList.json");
+
+    @Test
+    public void toModelType_typicalRecipesFile_success() throws Exception {
+        JsonSerializableRecipeList dataFromFile = JsonUtil.readJsonFile(TYPICAL_RECIPES_FILE,
+                JsonSerializableRecipeList.class).get();
+        RecipeList recipeListFromFile = dataFromFile.toModelType();
+        RecipeList typicalRecipesRecipeList = TypicalRecipes.getTypicalRecipeList();
+        assertEquals(recipeListFromFile, typicalRecipesRecipeList);
```
Comment (blackonyyx) [inline src/test/java/seedu/address/storage/JsonSerializableRecipeListTest.java:31]: bit of white space please :)

File: src/test/java/seedu/address/storage/StorageManagerTest.java [hunk 5]
```diff
@@ -11,22 +10,27 @@
...
 
     @BeforeEach
     public void setUp() {
-        JsonAddressBookStorage addressBookStorage = new JsonAddressBookStorage(getTempFilePath("ab"));
+        JsonItemListStorage itemListStorage = new JsonItemListStorage(getTempFilePath("ab"));
+        JsonRecipeListStorage recipeListStorage = new JsonRecipeListStorage(getTempFilePath("ac"));
+        JsonLocationListStorage locationListStorage = new JsonLocationListStorage(getTempFilePath("ad"));
         JsonUserPrefsStorage userPrefsStorage = new JsonUserPrefsStorage(getTempFilePath("prefs"));
-        storageManager = new StorageManager(addressBookStorage, userPrefsStorage);
+        storageManager = new StorageManager(itemListStorage, locationListStorage, recipeListStorage, userPrefsStorage);
```
Comment (blackonyyx) [inline src/test/java/seedu/address/storage/StorageManagerTest.java:34]: spacing

File: src/test/java/seedu/address/storage/JsonSerializableLocationListTest.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    private static final Path TEST_DATA_FOLDER = Paths.get("src",
+            "test", "data", "JsonSerializableLocationListTest");
+    private static final Path TYPICAL_LOCATIONS_FILE = TEST_DATA_FOLDER.resolve("typicalLocationsLocationList.json");
+    private static final Path INVALID_LOCATION_FILE = TEST_DATA_FOLDER.resolve("invalidLocationLocationList.json");
+    private static final Path DUPLICATE_LOCATION_FILE = TEST_DATA_FOLDER.resolve("duplicateLocationLocationList.json");
+
+    @Test
+    public void toModelType_typicalLocationsFile_success() throws Exception {
+        JsonSerializableLocationList dataFromFile = JsonUtil.readJsonFile(TYPICAL_LOCATIONS_FILE,
+                JsonSerializableLocationList.class).get();
```
Comment (blackonyyx) [inline src/test/java/seedu/address/storage/JsonSerializableLocationListTest.java:27]: spacing

File: src/test/java/seedu/address/storage/JsonSerializableItemListTest.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    private static final Path TYPICAL_ITEMS_FILE = TEST_DATA_FOLDER.resolve("typicalItemsItemList.json");
+    private static final Path INVALID_ITEM_FILE = TEST_DATA_FOLDER.resolve("invalidItemItemList.json");
+    private static final Path DUPLICATE_ITEM_FILE = TEST_DATA_FOLDER.resolve("duplicateItemItemList.json");
+
+    @Test
+    public void toModelType_typicalItemsFile_success() throws Exception {
+        JsonSerializableItemList dataFromFile = JsonUtil.readJsonFile(TYPICAL_ITEMS_FILE,
+                JsonSerializableItemList.class).get();
+        ItemList itemListFromFile = dataFromFile.toModelType();
+        ItemList typicalItemsItemList = TypicalItems.getTypicalItemList();
```
Comment (blackonyyx) [inline src/test/java/seedu/address/storage/JsonSerializableItemListTest.java:29]: spacing

File: src/test/java/seedu/address/testutil/TypicalItems.java [hunk 8]
```diff
@@ -14,11 +15,13 @@
 
     public static final Item APPLE = new ItemBuilder()
             .withId(1)
+            .withRecipe(new HashSet<>())
```
Comment (blackonyyx) [inline src/test/java/seedu/address/testutil/TypicalItems.java:18]: Minor observation but item stores as a list, so why are we using hashset in itemBuilder?

**Review by blackonyyx — APPROVED** [review]
LGTM, there are only mainly minor issues with spacing.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 1 above)
Comment (aqcd) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:19]: Replaced by the ItemParserUtil and RecipeParserUtil

File: src/test/java/seedu/address/logic/commands/ModelStub.java (hunk 3 above)
Comment (aqcd) [reply src/test/java/seedu/address/logic/commands/ModelStub.java:19]: Will be added when future tests require it, currently the item list is needed to reset the item's recipelist so some tests don't contaminate it and affect future tests.

File: src/test/java/seedu/address/testutil/TypicalItems.java (hunk 8 above)
Comment (aqcd) [reply src/test/java/seedu/address/testutil/TypicalItems.java:18]: The constructor takes Set in at the moment, might look into this in future

**Pull request merged**